core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -104,6 +108,52 @@ class HistoryServer(
 
   initialize()
 
+  private[spark] class HistoryServlet(val appId: String) extends HttpServlet {
+
+    // SPARK-5983 ensure TRACE is not supported
+    protected override def doTrace(req: HttpServletRequest, res: HttpServletResponse): Unit = {
+      res.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED)
+    }
+
+    protected override def doGet(req: HttpServletRequest, res: HttpServletResponse): Unit = {
+      try {
```

Comment:
I guess you should probably check the accept headers of the request, that they are OK w/ application/zip:
`req.getHeader(HttpHeaders.ACCEPT)`.  You could check w/ guava's `MediaType#is` (I couldn't find a simpler way ...)

```
val accepts = MediaType.parse(req.getHeader(HttpHeaders.ACCEPT))
if (MediaType.ZIP.is(accepts)) {
  ...
}
```